src/components/FindMovie/FindMovie.tsx

```
@@ -1,10 +1,73 @@
 import React from 'react';
 import './FindMovie.scss';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+
+interface Props {
+  movies: Movie[];
+  query: string;
+  loading: boolean;
+  currentMovie: Movie | undefined;
+  error: boolean;
+  setCurrentMovie: (value: Movie | undefined) => void;
+  setMovies: React.Dispatch<React.SetStateAction<Movie[]>>;
+  setQuery: (value: string) => void;
+  setLoading: (value: boolean) => void;
+  setError: (value: boolean) => void;
+}
+
+export const FindMovie: React.FC<Props> = ({
+  movies,
+  query,
+  loading,
+  currentMovie,
+  error,
+  setMovies,
+  setQuery,
+  setCurrentMovie,
+  setLoading,
+  setError,
+}) => {
+  const clear = () => {
+    setError(false);
+    setQuery('');
+    setCurrentMovie(undefined);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setLoading(true);
+
+    getMovie(query)
+      .then(movie => {
+        return movie;
+      })
+      .then(movie => {
+        if ('Error' in movie) {
+          setError(true);
+
+          return;
+        }
+
+        setCurrentMovie({
+          title: movie.Title,
+          description: movie.Plot,
+          imgUrl:
+            movie.Poster && movie.Poster !== 'N/A'
+              ? movie.Poster
+              : 'https://via.placeholder.com/360x270.png?text=no%20preview',
+          imdbUrl: `https://www.imdb.com/title/${movie.imdbID}`,
+          imdbId: movie.imdbID,
+        });
+      })
+      .catch(() => 'no movies found')
```

This `.catch()` block doesn't seem to do anything useful. It catches potential errors but its return value (`'no movies found'`) is not used. This can hide bugs by swallowing errors silently. Since API errors like "Movie not found" are already handled in the `.then()` block, this `catch` is likely unnecessary. It's better to either remove it or add proper logging (`console.error(error)`) to handle unexpected errors during the API call or processing.